### PR TITLE
GLFW example with dirty region management

### PR DESCRIPTION
**Pull request by betrevisan**
The changes proposed in this PR update the GLFW example to support dirty region management. It updates the example to a loading spinner in the center of the screen and adjusts the user-defined embedder callbacks to manage the information necessary for partial repaint. The goal of this example is also to serve as a local benchmark for dirty region management within the Embedder API.

Note: This PR is dependent on https://github.com/flutter/engine/pull/35022.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by bdero** [comment]
Given the additional complexity and build requirements not available on MacOS (EGL), can we make this a separate GLFW example? Perhaps under `examples/glfw_drm` with a `README.md` explaining what DRM is/pointing to the doc? It would be nice to keep `example/glfw` as the simplest practical demonstration for people just getting started with embedding Flutter.

**Comment by betrevisan** [comment]
> Given the additional complexity and build requirements not available on MacOS (EGL), can we make this a separate GLFW example? Perhaps under `examples/glfw_drm` with a `README.md` explaining what DRM is/pointing to the doc? It would be nice to keep `example/glfw` as the simplest practical demonstration for people just getting started with embedding Flutter.

I agree that it would be nice to keep the example as simple as possible. I will add the partial repaint example as an extra then.

**Review by bdero — COMMENTED** [review]
Just some comments about the build setup/running in CI and fetching extensions/doing proc resolution ahead of time. But overall looking good!

File: examples/glfw_drm/FlutterEmbedderGLFW.cc [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    PFNEGLSWAPBUFFERSWITHDAMAGEEXTPROC swap_buffers_with_damage_ = nullptr;
+    if (HasExtension(extensions, "EGL_EXT_swap_buffers_with_damage")) {
+      swap_buffers_with_damage_ =
+          reinterpret_cast<PFNEGLSWAPBUFFERSWITHDAMAGEEXTPROC>(
+              eglGetProcAddress("eglSwapBuffersWithDamageEXT"));
+    } else if (HasExtension(extensions, "EGL_KHR_swap_buffers_with_damage")) {
+      swap_buffers_with_damage_ =
+          reinterpret_cast<PFNEGLSWAPBUFFERSWITHDAMAGEEXTPROC>(
+              eglGetProcAddress("eglSwapBuffersWithDamageKHR"));
+    }
```
[the remark this answers is not in the document]
Comment (betrevisan) [reply examples/glfw_drm/FlutterEmbedderGLFW.cc:171]: Ah I didn't realize. I updated the code to store the surface and display as global variables and only update them once (before RunFlutter).

File: examples/glfw_drm/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    output_name = "embedder_example"
+
+    sources = [ "FlutterEmbedderGLFW.cc" ]
+
+    deps = [
+      "//flutter/shell/platform/embedder:embedder",
+      "//third_party/glfw",
+    ]
+
+    libs = [ "EGL" ]
```
[the remark this answers is not in the document]
Comment (betrevisan) [reply examples/glfw_drm/BUILD.gn:18]: Make sense. I updated the BUILD.gn file to only include the glfw_drm example as a dependency when not running mac_os.

File: examples/glfw_drm/BUILD.gn [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//flutter/examples/examples.gni")
+
+if (build_embedder_examples) {
+  executable("glfw") {
+    output_name = "embedder_example"
```
[the remark this answers is not in the document]
Comment (betrevisan) [reply examples/glfw_drm/BUILD.gn]: Sorry, I forgot to do this stage. Should be good now.

**Review by bdero — APPROVED** [review]

**Comment by bdero** [comment]
This looks like it should be good to go as soon as https://github.com/flutter/engine/pull/35022 lands

**Pull request merged**